### PR TITLE
- fixed and/or/not (replaced by C++ syntax)

### PR DESCRIPTION
**Pull request by Tigzy**
- fixes for Win32

### CODE REVIEW

File: ssdeep.h [hunk 1]
```diff
@@ -22,6 +22,7 @@
 
 // This is a kludge, but it works.
 #define __progname "ssdeep"
+#define VERSION "2.14.1"
```
Comment (a4lg) [inline ssdeep.h]: No, this is not acceptable (since `VERSION` macro is defined in "config.h").

File: match.cpp [hunk 2]
```diff
@@ -301,14 +301,14 @@ bool match_compare(state *s, Filedata * f)
     {
       if (!(_tcsncmp(f->get_filename(),
 		     (*it)->get_filename(),
-		     std::max(fn_len,_tcslen((*it)->get_filename())))) and
+		     max(fn_len,_tcslen((*it)->get_filename())))) &&
```
Comment (a4lg) [inline match.cpp]: Replacing `and` to `&&` is okay but is there any good reason to remove `std::`?

File: dig.cpp [hunk 3]
```diff
@@ -503,8 +503,8 @@ bool process_win32(state *s, TCHAR *fn)
     // original filename, e.g. C:\foo\*. When this happens it means we just
     // didn't find any matching files.
     // Note that we still display errors with the original 'fn'
-    if (not _tcsstr(fn, _TEXT("*")))
-      print_error_unicode(s, fn, "No such file or directory");
+    if (! _tcsstr(fn, _TEXT("*")))
+      print_error_unicode(s, fn, "No such file || directory");
```
[the remark this answers is not in the document]
Comment (Tigzy) [reply dig.cpp]: yes, this is a mistake.

File: ssdeep.h (hunk 1 above)
Comment (Tigzy) [reply ssdeep.h]: config.h doesn't exist, please make it available.

File: match.cpp (hunk 2 above)
Comment (Tigzy) [reply match.cpp]: For some reason it doesn't compile, maybe adding <algorithm.h> would fix?

**Comment by Tigzy** [comment]
I've added some fixes for mistakes I made.

File: ssdeep.h (hunk 1 above)
Comment (a4lg) [reply ssdeep.h]: Wait, are you trying to build ssdeep using Visual C++?

Comment (Tigzy) [reply ssdeep.h]: Yes, and I did. It works

**Comment by a4lg** [comment]
First of all, could you clean changes? For instance, you can (and should) separate one big commit to those three small commits for example:
1. Win32 "fix" (reviewing but seems to break other platform)
2. C++ syntax change (I think this is OK)
3. Other small changes (?)

Don't ask me how to do that because I don't have enough time to teach you.

**Comment by Tigzy** [comment]
What is it breaking? It's enclosed with proper _WIN32 preprocessor. 
I can redo the commits if you need, yes. Will open a new PR

File: ssdeep.h (hunk 1 above)
Comment (a4lg) [reply ssdeep.h]: OK, I respect you for making this working on Visual C++ but this line breaks everything other than your environment (normally, `configure` shell script generates `config.h` with proper definitions including `VERISON` and we use mingw-w64 to build Win32 binaries). So, this line is not acceptable. Sorry.

Comment (Tigzy) [reply ssdeep.h]: Ok, will enclose it with preprocessor then

Comment (a4lg) [reply ssdeep.h]: No, this is bad too (`_WIN32` is also defined on mingw-w64). Just not acceptable for now.

Comment (Tigzy) [reply ssdeep.h]: what about a config.h.win32 that has to be renamed for VC compilation?

**Comment by a4lg** [comment]
Uhh, I will accept your C++ and/or/not change (and maybe additional `#include <algorithm>`) but not other changes. I appreciate that you proved that ssdeep can be built on Visual C++ with minimal changes. So, I will plan proper support of Win32 (without breaking other platform).

**Comment by Tigzy** [comment]
Wait wait, I'm going to submit a new PR

**Comment by a4lg** [comment]
I merged your C++ coding style change as 57fcfffe1255c247c865c23cb139713e96679656.

**Comment by Tigzy** [comment]
Err, ok

**Pull request closed without merging**